profiles.wordpress.org/public_html/wp-content/plugins/wporg-5ftf-activity-handler/wporg-5ftf-activity-handler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/*
+Plugin Name: WordPress.org 5ftf Activity Handler
+Plugin URI: http://wordpress.org
+License: GPL2
+Version: 1.1
+Description: Handles saving of last 5ftf contribution.
+*/
+
+defined( 'ABSPATH' ) or die();
+
+if ( ! class_exists( 'WPOrg_5ftf_Activity_Handler' ) ) {
+
+	class WPOrg_5ftf_Activity_Handler {
+
+		/**
+		 * @var WPOrg_5ftf_Activity_Handler The singleton instance.
+		 */
+		private static $instance;
+
+		/**
+		 *  Name of meta key that tracks last contribution as a unix timestamp.
+		 *
+		 * @var string
+		 */
+		const last_contribution_meta_key = 'wporg_5ftf_last_contribution';
+
+		/**
+		 * Returns always the same instance of this plugin.
+		 *
+		 * @return WPOrg_5ftf_Activity_Handler
+		 */
+		public static function get_instance() {
+			if ( ! self::$instance ) {
+				self::$instance = new self();
+			}
+
+			return self::$instance;
+		}
+
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			if ( 'profiles.wordpress.org' === site_url() ) {
+				add_filter( 'bp_activity_add', array( $this, 'handle_wordpress_activity' ) );
+			} elseif ( str_ends_with( 'wordpress.org', site_url() ) ) {
+				add_filter( 'wporg_github_added_activity', array( $this, 'handle_github_activity' ) );
+			}
+		}
+
+		/**
+		 * Saves meta value if it qualifies as a contribution.
+		 */
+		public function handle_wordpress_activity( $args ) {
+			if ( self::is_5ftf_contribution( $args['type'] ) ) {
+				self::update_last_contribution_meta( $args['user_id'] );
+			}
+		}
+
+		/**
+		 * Saves meta value if it qualifies as a github contribution.
+		 *
+		 * $args.category string Type of github event
+		 * $args.repo string Name of the public repository
+		 * $args.user_id string|null Name of the public repository
+		 *
+		 * @return int|bool
+		 */
+		public function handle_github_activity( $args ) {
+			$valid_actions = array( 'pr_opened', 'pr_closed', 'pr_merged', 'issue_opened', 'pr_reopened' );
+
+			if ( empty( $args['user_id'] ) ) {
+				// user_id may be null if the user didn't connect their github account to their wordpress.org account
+				return '-1 WordPress.org user id is missing';
+			}
+
+			if ( ! in_array( $args['category'], $valid_actions, true ) ) {
+				return '-1 Category: ' . sanitize_text_field( $args['category'] ) . ' is not a contribution.';
+			}
+
+			return self::update_last_contribution_meta( $args['user_id'] );
+		}
+
+		/**
+		 * Returns whether action is considered a contribution.
+		 *
+		 * @return boolean True if action == contribution
+		 */
+		public function is_5ftf_contribution( $action ) {
```

Comment:
It might be good to clarify the name of this function and/or `handle_github_activity()`, since the name here implies that it'd also be used for GitHub/Trac/etc.

Or maybe `handle_github_activity()` should call this function, to be consistent w/ `handle_wordpress_activity()`? Or the contents of this function moved into `handle_wordpress_activity()`?

I don't feel strongly which one, as long as the end result is consistent/clear.

Reply:
Yeah I agree, I feel like until I run into handling specific cases, which will definitely happen soon, I'm not sure what convention will work best.

 I'm in a bit of a spot with this PR where I'm thinking:

1. Get this code in production early and have it working (even if the value is not user), then follow up with the more specific cases and refactoring
2. Wait for more events or clearer definitions for teams, refactor, then deploy.

I'm leaning towards #1. Get something up and iterate as more data sources are introduced or team activities are better solidified. 

When I look at the team-specific tickets in the [non-code milestone](https://github.com/WordPress/five-for-the-future/milestone/8) none of them are complete yet or have all their discussions resolved. I don't think I want this PR hanging around that long unless we think we'll never get to use this meta value.

Thoughts?

Reply:
Yeah, `#1` sounds like the better option to me too 👍🏻